pkg/cloudprovider/providers/gce/gce_loadbalancer.go

```diff
@@ -37,7 +37,14 @@ import (
 
 // GetLoadBalancer is an implementation of LoadBalancer.GetLoadBalancer
 func (gce *GCECloud) GetLoadBalancer(clusterName string, service *v1.Service) (*v1.LoadBalancerStatus, bool, error) {
-	loadBalancerName := cloudprovider.GetLoadBalancerName(service)
+	// For backword compatibility we had to fetch the loadBalancerName this way or else a simple assignment for annotations["currentLB"] would be enough.
+	annotations := service.GetAnnotations()
+	var loadBalancerName string
+	if value, ok := annotations["currentLB"]; ok {
```

Comment:
the annotation shoud a) be a const somewhere and b) be namespaced.  Something like `service.kubernetes.io/load-balancer-name`